include/libtorrent/aux_/session_impl.hpp

```diff
@@ -118,8 +118,40 @@ namespace libtorrent
 		class item;
 	}
 
-	struct listen_socket_t
+	struct session_udp_socket : utp_socket_interface
 	{
+		session_udp_socket(io_service& ios)
+			: sock(ios) {}
+
+		virtual udp::endpoint local_endpoint() override { return sock.local_endpoint(); }
```

Comment:
The cppCoreGuidelines recommends using only one of "virtual" or "override". Since "override" implies virtual. In fact, I would have thought cpplint.py would have flagged this